Source/Core/InputCommon/ControllerInterface/XInput/XInput.h

```diff
@@ -90,7 +90,7 @@ class Device : public Core::Device
 
 private:
 	XINPUT_STATE m_state_in;
-	XINPUT_VIBRATION m_state_out;
+	XINPUT_VIBRATION m_state_out, m_current_state_out;
```

Comment:
> I don't know for sure if the constructor might be called 2x

Why would this be relevant? Either way you're zeroing out the struct's members. Also you can't call the constructor twice on an object (aside from abusing placement new, which we don't use for this).

> And XINPUT_STATE m_state_in should also get the same treatment, which is out of scope for the pr.

Sorry, but no. Just because something is done in a less than desirable way does not mean it's a free pass to add additional code onto it.